src/NotificationTarget.php

```diff
@@ -543,18 +543,23 @@ public function addToRecipientsList(array $data)
         if (isset($data['name']) && !empty($data['name'])) {
             $username = $data['name'];
         }
+
         if (isset($data['users_id']) && ($data['users_id'] > 0)) {
             $user = new User();
             if (
                 !$user->getFromDB($data['users_id'])
                 || ($user->getField('is_deleted') == 1)
+                // if notification not send by NotificationTargetCommonITILObject
+                // check if user explicitly refused notification
+                //for CommonITILObject use_notification is check before
+                || (!$user->isUserNotificationEnable() && !is_a(get_called_class(), NotificationTargetCommonITILObject::class, true))
```

Comment:
Same remark here. We should rely only on `NotificationTarget::validateSendTo()` to filter recipients.
```suggestion
```

Reply:
done